File: web/src/services/models/Company.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

import type { Employee } from './Employee';

/**
 * Models a Company entity.
 */
export type Company = {
    /**
     * The ID of the entity in the MongoDB
     */
    id?: string | null;
    /**
     * The name associated with the entity in the MongoDB
     */
    label?: string | null;
    /**
     * The address of the company.
     */
    address?: string | null;
    /**
     * The URL of the website for the given company.
     */
    webUrl?: string | null;
    /**
     * The list of Employees associated with this Company.
     */
    employees?: Array<Employee> | null;
}

File: api/Domain/Model/Company.cs
namespace Api.Domain.Model;

/// <summary>
/// Models a Company entity.
/// </summary>
public class Company : MongoEntityBase
{
    /// <summary>
    /// The address of the company.
    /// </summary>
    public string? Address { get; set; }

    /// <summary>
    /// The URL of the website for the given company.
    /// </summary>
    public string? WebUrl { get; set; }

    /// <summary>
    /// The list of Employees associated with this Company.
    /// </summary>
    /// <remarks>
    /// To efficiently store this, we want to map the Company ID on the Employee instead
    /// of mapping the Employee ID on the Company.  This is because a Company will have
    /// many Employees while an Employee will have typically 1 but perhaps more Companies
    /// so it is generally more efficient to "lookup all Employees for a given Company ID"
    /// than to "lookup all Employee IDs for a given Company".  So we want to ignore this
    /// on save.
    /// </remarks>
    [BsonIgnore]
    [MongoLookup]
    public IEnumerable<Employee> Employees { get; set; } = new List<Employee>();
}